### PR TITLE
Feat page my events

### PR DESCRIPTION
**Pull request by DiLiUA**
- change launchEvent inteface 
        - responcive markup for my events page
        - created show-info component
        - create statistics component
        - aded functionality for liking/unliking show

### CODE REVIEW

File: src/event-launch/event-launch.component.ts [hunk 1]
```diff
@@ -59,31 +59,39 @@ export class LaunchComponent implements OnInit, OnDestroy {
     });
 
     this.launchEvent = {
```
Comment (valorkin) [inline src/event-launch/event-launch.component.ts]: Use class with default values

File: src/shared/search-service/search.service.ts [hunk 2]
```diff
@@ -43,6 +43,14 @@ export class SearchService {
     });
   }
 
+  public getMyEvents(query: string): Observable<any> {
+    return this.http.get(`${Config.api}/get-my-events?${query}`)
+      .map((res: any) => {
```
Comment (valorkin) [inline src/shared/search-service/search.service.ts]: Single resposibility
One data transformation per map call
Dont forget ti catch errors

File: src/shared/statistics/statistics.component.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+@Component({
+  selector: 'app-statistics',
+  templateUrl: 'statistics.component.html',
+  styleUrls: ['../../my-events/styles.scss', 'statistics.component.scss']
+})
+export class StatisticsComponent implements OnInit {
+  @Input() statistics: Statistics;
+  @Input() id: string;
+  @Input() isEvent = true;
+  user: any;
```
Comment (valorkin) [inline src/shared/statistics/statistics.component.ts]: Don't use any, provide appropriate classes

File: src/shared/statistics/statistics.component.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  constructor(private statisticsService: StatisticsService) {
+
+  }
+
+  ngOnInit() {
+    this.user = JSON.parse(localStorage.getItem('profile'));
```
Comment (valorkin) [inline src/shared/statistics/statistics.component.ts]: Any json.parse should be wrapped in try catch

File: src/shared/statistics/statistics.component.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  }
+
+  setLike(): void {
+    if (this.isEvent) {
```
Comment (valorkin) [inline src/shared/statistics/statistics.component.ts]: Invert if

File: src/shared/statistics/statistics.service.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+import { LikeRequestObj } from './statistics.interface';
+import { Config } from '../../app.config';
+
+@Injectable()
+export class StatisticsService {
+
+  constructor(private http: Http) {
+  }
+
+  private setOptions(): RequestOptions {
```
Comment (valorkin) [inline src/shared/statistics/statistics.service.ts]: Its get options

File: src/shared/statistics/statistics.service.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+      headers: new Headers({
+        'Content-Type': 'application/json',
+        'Accept': 'application/json',
+      })
+    });
+  }
+
+  setShowLike(rqstObg: LikeRequestObj): Observable<any> {
+    return this.http.post(`${Config.api}/show-like`, JSON.stringify(rqstObg),  this.setOptions())
+      .map(res => {
```
Comment (valorkin) [inline src/shared/statistics/statistics.service.ts]: Use one line arrow func and single responsibility

File: src/shared/statistics/statistics.service.ts [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+        return res.json().data;
+      }, (err: any) => {
```
Comment (valorkin) [inline src/shared/statistics/statistics.service.ts]: Catch

Comment (valorkin) [reply src/shared/statistics/statistics.service.ts]: Review your PR and apply same to other code

**Review by valorkin — CHANGES_REQUESTED** [review]

File: src/shared/statistics/statistics.service.ts [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+        console.log(err);
+      });
+  }
+
+  setArtistLike(rqstObg: LikeRequestObj): Observable<any> {
+    return this.http.post(`${Config.api}/artist-like`, JSON.stringify(rqstObg),  this.setOptions())
+      .map(res => {
+        console.log('artist Like: ', res);
```
Comment (Ugmaxie) [inline src/shared/statistics/statistics.service.ts]: remove console.log

**Review by Ugmaxie — CHANGES_REQUESTED** [review]

**Review by Ugmaxie — APPROVED** [review]

**Comment by Ugmaxie** [comment]
@valorkin Your comments seems to be applied. Merging.

**Pull request merged**